tests/pytests/test_numbers.py

```diff
@@ -389,3 +389,69 @@ def testNegativeValues(env):
     # Query the index. if the split value of the root is nan, the query won't return any results.
     res = env.cmd('FT.SEARCH', 'idx', '@num:[-inf +inf]', 'NOCONTENT')
     env.assertEqual(res[0], doc_id)
+
+def testNumberFormat(env):
+    env = Env(moduleArgs = 'DEFAULT_DIALECT 2')
+    conn = getConnectionByEnv(env)
+
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 'n', 'numeric').ok()
+    env.assertEqual(conn.execute_command('HSET', 'doc01', 'n', '1.0'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc02', 'n', '1'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc03', 'n', '1.0e0'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc04', 'n', '10e+2'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc05', 'n', '1.5e+2'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc06', 'n', '10e-2'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc07', 'n', '1.5e-2'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc08', 'n', 'INF'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc09', 'n', 'inf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc10', 'n', 'iNf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc11', 'n', '+INF'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc12', 'n', '+inf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc13', 'n', '+iNf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc14', 'n', '-INF'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc15', 'n', '-inf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc16', 'n', '-iNf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc17', 'n', '-1'), 1)
+
+    # Test unsigned numbers
+    expected = [3, 'doc01', ['n', '1.0'], 'doc02', ['n', '1'],
+                'doc03', ['n', '1.0e0']]
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[1 1]')
```

Comment:
Consider simplifying using `NOCONTENT` (also in other occurrences)

Reply:
OK. Done. Thanks